### ISSUE TITLE
Kratos program is not working

### ISSUE THREAD

**Issue by matheusoliveira202** [KratosMultiphysics/Kratos#9104]
**Description**
Hello good Morning
The program is not working correctly, as the Kratos 8.0.1 extension cannot compile any code, causing the following error attached. I can't compile the codes that were already working and not even when running the examples that are in the library included in the program.
thanks in advance

**Scope**
Where in Kratos is this bug happening. 
- Compilation

**To Reproduce**
Steps to reproduce the bug (if apply). Add any custom example files here if relevant.
---
Traceback (most recent call last):
  File "MainKratos.py", line 33, in <module>
    DEMAnalysisStageWithFlush(model, project_parameters).Run()
  File "C:\PROGRA~1\GiD\GID150~1.3\PROBLE~1\kratos.gid\exec\kratos\KratosMultiphysics\analysis_stage.py", line 49, in Run
    self.Initialize()
  File "C:\PROGRA~1\GiD\GID150~1.3\PROBLE~1\kratos.gid\exec\kratos\KratosMultiphysics\DEMApplication\DEM_analysis_stage.py", line 264, in Initialize
    self.ReadModelParts()
  File "C:\PROGRA~1\GiD\GID150~1.3\PROBLE~1\kratos.gid\exec\kratos\KratosMultiphysics\DEMApplication\DEM_analysis_stage.py", line 446, in ReadModelParts
    self.ReadModelPartsFromMdpaFile(max_node_id, max_elem_id, max_cond_id)
  File "C:\PROGRA~1\GiD\GID150~1.3\PROBLE~1\kratos.gid\exec\kratos\KratosMultiphysics\DEMApplication\DEM_analysis_stage.py", line 418, in ReadModelPartsFromMdpaFile
    ReadModelPart(self.spheres_model_part, self.GetDiscreteElementsInputFileTag(), max_node_id, max_elem_id, max_cond_id)
  File "C:\PROGRA~1\GiD\GID150~1.3\PROBLE~1\kratos.gid\exec\kratos\KratosMultiphysics\DEMApplication\DEM_analysis_stage.py", line 399, in ReadModelPart
    model_part_io = self.model_part_reader(file_path, max_node_id, max_elem_id, max_cond_id)
  File "C:\PROGRA~1\GiD\GID150~1.3\PROBLE~1\kratos.gid\exec\kratos\KratosMultiphysics\DEMApplication\DEM_analysis_stage.py", line 59, in model_part_reader
    return ReorderConsecutiveFromGivenIdsModelPartIO(modelpart, nodeid, elemid, condid, IO.SKIP_TIMER)
RuntimeError: Error: Error opening mdpa file : C:\Users\proje\OneDrive\Área de Trabalho\wEFwf.gid\wEFwfDEM.mdpa

in kratos/sources/model_part_io.cpp:64:ModelPartIO::ModelPartIO
---
**Expected behavior**
A clear and concise description of what should be the expected behaviour.

**Environment**
- Win64 ( Ram 16 / AMD Ryzen 7 4700G Radeon Graphics 3.60 Ghz)
- GiD v15.0.3
- Kratos 8.1.0 version

I would also take the opportunity to ask if there is any material available (tutorial), in addition to the manuals, that would help me in the realization of a model, in which triaxis tests will be carried out, on a cilytic sample of particles surrounded by a membrane. I sincerely thank you for any help you can make available.
Thanks also for the promptness of the answers

**Comment by rubenzorrilla** [comment]
Referring to the @KratosMultiphysics/dem team.

**Comment by salvalatorre** [comment]
Thanks, Matheus, I'm looking into it.
Please, to begin with, try in general not to use paths having spaces or accents...

**Comment by matheusoliveira202** [comment]
Thanks, @rubenzorrilla  @salvalatorre @maceligueta @GuillermoCasas 
Thanks in advance for all the help

**Comment by AlejandroCornejo** [comment]
Is it solved? :)

**Comment by salvalatorre** [comment]
It is about to. More than one working on it... @jginternational

**Comment by maceligueta** [comment]
Hello @matheusoliveira202 , thanks for reporting the error. It was very helpful that you already posted the release version you used and the GiD version you used. 
It is quite strange that the input file (mdpa file) could not be found. Maybe it was not generated correctly and then it couldn't be found. Didn't you find any other error within GiD before this one? If you could send the whole example as a zip file it would be very helpful to debug it.
Concerning the creation of triaxial tests, we used to have an option to set all the boundary conditions automatically, but now it is not available. We can work on it (with the help of @jginternational if possible). Actually, we do not use a membrane, but we put a lateral pressure instead. The sample is compressed by two plates and the same pressure is applied to the lateral walls, until the target confinement stress is reached. Then, the plates keep pushing while the lateral pressure is kept constant. 
Let's first fix the error and then we will add the Triaxial Test option.

**Comment by maceligueta** [comment]
It seems I was wrong, and the Cohesive DEM it does have the option to create a Triaxial test under General Settings -> Material Analysis. However, the most recent tests have revealed a bug which we are trying to fix. Some kind of inconsistency between the GUI and Kratos DEM. We are working on it.

**Comment by matheusoliveira202** [comment]
good afternoon , thank you very much!!

Holla, thanks for everyone's support ...The bug regarding compilation was resolved, just switching to a directory without spacing and accents was enough to get them to work.
Thank you very much @salvalatorre  ...
----
However, I am really having difficulties regarding the implementation of this triaxial test option, I have been testing it but the confinement pressure does not seem to be active which is allowing the particles to flow, without maintaining the integrity necessary for a test simulation.

I will attach the file already with the properties of the test piece and the assignments of displacements of the particles in the compression planes, as it can be noticed even activating the Triaxial test under in General Settings -> Material Analysis. This has not been attributed (I have tested for a range of confining pressure values and had no effect)
... I do not know if it will be necessary to close this forum and open another one so that I can resolve this issue , but once again I thank you all for all your help .

**Comment by matheusoliveira202** [comment]
> It seems I was wrong, and the Cohesive DEM it does have the option to create a Triaxial test under General Settings -> Material Analysis. However, the most recent tests have revealed a bug which we are trying to fix. Some kind of inconsistency between the GUI and Kratos DEM. We are working on it.

**Comment by matheusoliveira202** [comment]
@maceligueta 
I can't add files in Zip format, Rar .. which formats would be accepted on the platform.
![printaaddfile](https://user-images.githubusercontent.com/90060599/132721239-5dd1cfbd-0a3c-401e-bba8-78ce1e214a42.jpg)

follow a link to a google folder
 https://drive.google.com/drive/folders/1oNunumeOEzkW7b7fvdUf9KOJVtCMuaYH?usp=sharing

**Comment by maceligueta** [comment]
Hello @matheusoliveira202 . We are about to finish fixing the code so it can run triaxial tests easily. However, as I understood, you downloaded a pre-compiled version of Kratos, right? You did not compile it on your PC.
I am thinking of some way to introduce the new goodies in a fixed release version. What's the usual procedure, @roigcarlo ? Should we wait for the next Release, or there's a possibility of launching intermediate releases?

**Comment by roigcarlo** [comment]
In general if we want to introduce fixes to an existing version we launch a new minor 8.1.1 or 8.2 if the changes are relevant enough.
Having in account that we will release 9.0 (and accounting that we will change all release mechanism etc...) by the end of October I would suggest to wait until then. If the problem can't be solved or you can' wait tell me and I will try to generate a new release with the changes you need.

Also @matheusoliveira202 sorry but github does not accept big compressed files directly into issue messages. This is a policy from Github, not Kratos, and there is nothing we can do, if you want to attach files it needs to be done as you did through drive, dropbox, etc...

**Comment by matheusoliveira202** [comment]
Hello good Morning.  @maceligueta @roigcarlo ..I can't tell you if this version is a pre-compiled version of kratos 8.1.0 , but it's the one I found available through the gid platform. If there is another version available for me to try to do my essays I would be very grateful because, I have a certain urgency, to obtain these results, as it is part of the component of my graduate project. In which results are expected to be publishable.

In the drive folder there is a file containing all the characteristics of the material that I am working on, I would like to be able to do triaxial tests for confining pressures of 50, 100, 200 and 300 kPa
I don't understand, I can't compile and I get this error message
-----
Traceback (most recent call last):
  File "MainKratos.py", line 33, in <module>
    DEMAnalysisStageWithFlush(model, project_parameters).Run()
  File "C:\PROGRA~1\GiD\GiD 15.0.3\problemtypes\kratos.gid\exec\kratos\KratosMultiphysics\analysis_stage.py", line 49, in Run
    self.Initialize()
  File "C:\PROGRA~1\GiD\GiD 15.0.3\problemtypes\kratos.gid\exec\kratos\KratosMultiphysics\DEMApplication\DEM_analysis_stage.py", line 300, in Initialize
    self.materialTest.Initialize(self.DEM_parameters, self.procedures, self._GetSolver(), self.graphs_path, self.post_path, self.spheres_model_part, self.rigid_face_model_part)
  File "C:\PROGRA~1\GiD\GiD 15.0.3\problemtypes\kratos.gid\exec\kratos\KratosMultiphysics\DEMApplication\DEM_procedures.py", line 1223, in Initialize
    self.script.Initialize()
  File "C:\PROGRA~1\GiD\GiD 15.0.3\problemtypes\kratos.gid\exec\kratos\KratosMultiphysics\DEMApplication\DEM_material_test_script.py", line 95, in Initialize
    self.PrepareTests()
  File "C:\PROGRA~1\GiD\GiD 15.0.3\problemtypes\kratos.gid\exec\kratos\KratosMultiphysics\DEMApplication\DEM_material_test_script.py", line 172, in PrepareTests
    initial_height_top = y_top_total/weight_top
ZeroDivisionError: float division by zero

-----
Once again
I appreciate all the support you have given me
The only problem is the confining that is not being applied to complete this task. I am very grateful for any help that may have come.

**Comment by roigcarlo** [comment]
Hi @matheusoliveira202, if you obtained it with GiD is the precompiled, so no compilation involved. 
The error seems to be a division by zero so `weight_top` is 0. Not sure why is the reason. 

@maceligueta are the fixes python only? If so i can try to make a repack fast and send it to @matheusoliveira202. If the fix has changes in the cpp tell me and I will recompile and make a pack for him.

**Comment by salvalatorre** [comment]
Fixes are python only, so we could try that repack out!

**Comment by salvalatorre** [comment]
We have also had some Tcl and Tk modifications, but in the master codes, maybe those changes are not needed for the precompiled version...

**Comment by roigcarlo** [comment]
@salvalatorre if you can send me the scripts I need to replace I will prepare the pack. The TCL changes are part of the GiD problemtype, I can't help with that.

**Comment by jginternational** [comment]
To get the latest GUI version I recommend Git pull, but if you are not into git.
get a zip with the master branch

![image](https://user-images.githubusercontent.com/5918085/133093707-eeefec4c-41d0-48c8-a12b-efc82529b98d.png)

**Comment by maceligueta** [comment]
The address is this one: https://github.com/KratosMultiphysics/GiDInterface 
We fixed many things in the Graphical User Interface and also in the computing core (Kratos). If you are able to compile the last version of Kratos and download the last version of the GUI, it should work. 
Please let us know if you manage to do these two things. Otherwise we will try to pack a version for you for direct download, but this takes some work. First option is the best, because you can keep updating the code with the latest bugfixes and features we will be adding.

**Comment by matheusoliveira202** [comment]
Hello good Morning @maceligueta thank you but once for your attention and promptness,
I'm installing this version of Developer Gid (GiD 15.1.2d), to be quite honest I'm having a bit of a hard time understanding these link-linking steps.
----
4- Navigate to kratos.gid / exec /
There create a symbolic link to the Kratos installation folder (where runkratos is located)
Windows: mklink / J Kratos C: \ Kratos or perhaps mklink / J Kratos C: \ kratos \ bin \ Release (choose the actual Kratos installation folder)
__

I can't find the file that should replace this path... I did the procedure as I did before in the previous version, copying and pasting the Kratos.gid folder into the path
C: \ Program Files \ GiD \ GiD 15.1.2d \ problemtypes

however, when running the model, the same error message of division by zero is occurring when I activate Material Analysis.

Traceback (most recent call last):
  File "MainKratos.py", line 33, in <module>
    DEMAnalysisStageWithFlush(model, project_parameters).Run()
  File "C:\PROGRA~1\GiD\GiD 15.1.2d\problemtypes\kratos.gid\exec\kratos\KratosMultiphysics\analysis_stage.py", line 49, in Run
    self.Initialize()
  File "C:\PROGRA~1\GiD\GiD 15.1.2d\problemtypes\kratos.gid\exec\kratos\KratosMultiphysics\DEMApplication\DEM_analysis_stage.py", line 300, in Initialize
    self.materialTest.Initialize(self.DEM_parameters, self.procedures, self._GetSolver(), self.graphs_path, self.post_path, self.spheres_model_part, self.rigid_face_model_part)
  File "C:\PROGRA~1\GiD\GiD 15.1.2d\problemtypes\kratos.gid\exec\kratos\KratosMultiphysics\DEMApplication\DEM_procedures.py", line 1223, in Initialize
    self.script.Initialize()
  File "C:\PROGRA~1\GiD\GiD 15.1.2d\problemtypes\kratos.gid\exec\kratos\KratosMultiphysics\DEMApplication\DEM_material_test_script.py", line 95, in Initialize
    self.PrepareTests()
  File "C:\PROGRA~1\GiD\GiD 15.1.2d\problemtypes\kratos.gid\exec\kratos\KratosMultiphysics\DEMApplication\DEM_material_test_script.py", line 172, in PrepareTests
    initial_height_top = y_top_total/weight_top
ZeroDivisionError: float division by zero

I believe must be making a mistake; in the installation process or even in the model, because even this version presents the same message, and for me, it would be very important to carry out these triaxial and hydrostatic tests for confining pressures of 50, 100, 200 and 300 kPa until the end of this month.
But once, thanks for all the support and attention
